src/python/essentia/pyutils/extractors.py

```diff
@@ -0,0 +1,141 @@
+# Copyright (C) 2006-2016  Music Technology Group - Universitat Pompeu Fabra
+#
+# This file is part of Essentia
+#
+# Essentia is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Affero General Public License as published by the Free
+# Software Foundation (FSF), either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the Affero GNU General Public License
+# version 3 along with this program. If not, see http://www.gnu.org/licenses/
+
+from argparse import ArgumentParser
+from multiprocessing import Pool
+from subprocess import Popen, PIPE
+
+import numpy as np
+import os
+import sys
+
+
+def __subprocess__(cmd):
+    """
+    General purpose subprocess.
+    """
+
+    cmd_str = ' '.join(cmd)
+    print('Running "{}"...'.format(cmd_str))
+
+    stdout, stderr = Popen(cmd, stdout=PIPE, stderr=PIPE).communicate()
+
+    stdout = stdout.decode("utf-8")
+    stderr = stderr.decode("utf-8")
+
+    stderr = 'cmd: "{}"\nstderr: "{}"'.format(cmd_str, stderr)
+    stdout = 'cmd: "{}"\nstderr: "{}"'.format(cmd_str, stdout)
+    return stdout, stderr
+
+
+def batch_music_extractor(audio_dir, output_dir, generate_log=True,
+                          audio_types=None, profile=None,
+                          store_frames=False, skip_analyzed=False, jobs=4):
```

Comment:
We can [detect the number of jobs automatically](https://docs.python.org/2/library/multiprocessing.html#multiprocessing.cpu_count) if that makes sense.